docs/design/2018-10-24-view-support.md

```diff
@@ -5,65 +5,123 @@
-I implement the "`CREATE OR REPLACE`" command to create a view with the `SELECT` statement. Currently, I reuse "`DROP TABLE`" to drop a view, and "`SHOW FULL TABLES`" can also display the view list with different table types. The "`SHOW CREATE TABlE`" command can also regenerate a view create command.
+This proposal is prepared to implement basic VIEW feature, which contains "CREATE OR REPLACE VIEW", "SELECT FROM VIEW", "DROP VIEW" and "SHOW TABLE STATUS". All other unimplemented feature will list as compatibility and discuss later.
+and we mainly introduce new struct named `ViewInfo` to store view's metadata.
 
 ## Rationale
-`VIEW` is just a tableinfo object store with a `SELECT` statement, and only supports two types of DDL operations, which are "`CREATE OR REPLACE VIEW`" and "`DROP`". Currently, view can only be used within the `SELECT` statement.  
+`VIEW` is just a `TableInfo` object store with a `ViewInfo` struct, and only supports limited DDL operations.  
+Here is `ViewInfo`'s attributes and detail attribute explaination:
+```
+type ViewInfo struct {
+	Algorithm   AlgorithmType    `json:"view_algorithm"`
+	Definer     string           `json:"view_definer"`  
+	Security    SecurityType     `json:"view_security""`
+	SelectStmt  string           `json:"view_select"`
+	CheckOption CheckOption      `json:"view_checkoption"`
+	Cols        []string         `json:"view_cols"`
+	IsUpdatable bool             `json:"view_isUpdatable"`
+}
+```
+* AlgorithmType  ref https://dev.mysql.com/doc/refman/5.7/en/view-algorithms.html   
+    The view SQL AlGORITHM characteristic. The value is one of UNDEFINED、MERGE OR TEMPTABLE, if no ALGORITHM clause is present, UNDEFINED is the default algorithm.
+    We will implement Algorithm=UNDEFINED only now.
+* Definer  ref https://dev.mysql.com/doc/refman/5.7/en/create-view.html  
+    The account of the user who created the view, in 'user_name'@'host_name' format.
+* SecurityType  ref https://dev.mysql.com/doc/refman/5.7/en/create-view.html  
+    The view SQL SECURITY characteristic. The value is one of DEFINER or INVOKER.
+* CheckOptionType  ref https://dev.mysql.com/doc/refman/5.7/en/view-check-option.html  
+    The WITH CHECK OPTION clause can be given for an updatable view to prevent inserts to rows for which the WHERE clause in the select_statement is not true. It also prevents updates to rows for which the WHERE clause is true but the update would cause it to be not true (in other words, it prevents visible rows from being updated to nonvisible rows).  
+    In a WITH CHECK OPTION clause for an updatable view, the LOCAL and CASCADED keywords determine the scope of check testing when the view is defined in terms of another view. When neither keyword is given, the default is CASCADED.
+* IsUpdatable  ref https://dev.mysql.com/doc/refman/5.7/en/view-updatability.html  
+    This parameter mark if this view is updatable.
+* SelectStmt
+    This string is the select sql statement after sql rewriter.
```

Comment:
1. We only store the origin sql of create_view now.
2. Add an example here to show the expected format.